--- a/src/AIR2STREAM_READ.f90
+++ b/src/AIR2STREAM_READ.f90
@@ -1,7 +1,7 @@
 SUBROUTINE read_calibration

 USE commondata
-USE ifport                          ! necessary for makedirqq
+!USE ifport                          ! necessary for makedirqq

 IMPLICIT NONE
 INTEGER:: i, j, status
@@ -32,7 +32,7 @@
 WRITE(string,'(i1)' ) version

 folder = TRIM(name)//'/output_'//string//'/'
-result=makedirqq(folder)
+CALL EXECUTE_COMMAND_LINE('mkdir -p ' // TRIM(folder))
 WRITE(*,*) 'Objective function ',fun_obj

 IF (runmode .eq. 'FORWARD') THEN
@@ -90,8 +90,8 @@
     ! write parameters
     OPEN(unit=2,file=TRIM(folder)//'/parameters.txt',status='unknown',action='write')
     WRITE(2,'(I2,A)') n_par, '   !numero parametri'
-    WRITE(2,'(<n_par>(F10.5,1x))') (parmin(i),i=1,n_par)
-    WRITE(2,'(<n_par>(F10.5,1x))') (parmax(i),i=1,n_par)
+    WRITE(2,*)  (parmin(i),i=1,n_par)
+    WRITE(2,*)  (parmax(i),i=1,n_par)
     CLOSE(2)

 END IF
--- a/src/AIR2STREAM_RUNMODE.f90
+++ b/src/AIR2STREAM_RUNMODE.f90
@@ -41,7 +41,7 @@
 ALLOCATE (r(2*n_par),gbest(n_par),fit(n_particles),fitbest(n_particles))

 ! open file for the writing of all parameter set + efficiency index
-OPEN(unit=10,file=TRIM(folder)//'/0_'//TRIM(runmode)//'_'//fun_obj//'_'//TRIM(station)//'_'//series//'_'//TRIM(time_res)//'.out',status='unknown',action='write',form='binary')
+OPEN(unit=10,file=TRIM(folder)//'/0_'//TRIM(runmode)//'_'//fun_obj//'_'//TRIM(station)//'_'//series//'_'//TRIM(time_res)//'.out',status='unknown',action='write',form='unformatted',access='stream')

 x=0; v=0;
 r=0
@@ -191,7 +191,7 @@
 foptim=-999

 ! open file for the writing of all parameter set + efficiency index
-OPEN(unit=10,file=TRIM(folder)//'/0_'//TRIM(runmode)//'_'//fun_obj//'_'//TRIM(station)//'_'//series//'_'//TRIM(time_res)//'.out',status='unknown',action='write',form='binary')
+OPEN(unit=10,file=TRIM(folder)//'/0_'//TRIM(runmode)//'_'//fun_obj//'_'//TRIM(station)//'_'//series//'_'//TRIM(time_res)//'.out',status='unknown',action='write',form='unformatted',access='stream')

 CALL random_seed()

--- a/src/AIR2STREAM_SUBROUTINES.f90
+++ b/src/AIR2STREAM_SUBROUTINES.f90
@@ -379,7 +379,7 @@
	WRITE(*,*) 'Controllo superato'
 END IF

-WRITE(11,'(<n_par>(f10.6,1x))') (par_best(i),i=1,n_par)
+WRITE(11,*)  (par_best(i),i=1,n_par)
 WRITE(11,'(f10.6)') ei_check

 OPEN(UNIT=12,FILE=TRIM(folder)//'/2_'//TRIM(runmode)//'_'//fun_obj//'_'//TRIM(station)//'_'//series//'c_'//TRIM(time_res)//'.out',STATUS='unknown',ACTION='write')
